React Website Revamp: First Iteration

julientregoat opened the pull request:
Here is the first iteration of the React based revamp of the website. 

As we discussed on our call last week, the goal for this first draft was to primarily to recreate the original style, with a couple of changes and updates. I'm going to come up with a roadmap with milestones to break down the long term update of the website into smaller tasks. I'll also start up the project board here on Github shortly. 

### Timeline

**First iteration**

- Finish initial remake of site, displaying same content as current site, for use
- [x] Animations
- [x] Social Links
- [x] Styling
- [x] Mobile Views
---> Goal:  4/28/18

**Upcoming iterations**
- Pull in content from other feeds
---> Medium posts displayed as posted
---> Gitter feed
---> Twitter feed
---> What to emphasize with media and fetching, and what not?
---> Goal: 5/5/18 (likely done earlier)
- Add multisig donation address
---> Goal: 5/5/18

rauljordan commented:
Thanks for putting this together, Julien! Let us know when you think the changes are ready for review.

prestonvanloon commented:
Please update to serve properly through git pages.
I think we must have an `index.html` in the root directory or we can configure it to serve from `/docs` directory.

julientregoat commented:
@prestonvanloon Yep, that directory was just a few files from the old site and some of the SVGs I was working with. I'll remove that.

julientregoat commented:
Okay, I added a script within the `package.json` file to build the app into the a `docs/` directory.  It also adds the CNAME in there.

prestonvanloon commented:
Can you add a screenshot? When I ran the `/docs` directory with a simple http server, this is what I saw: 
![image](https://user-images.githubusercontent.com/7246818/38789498-9816fcbc-4108-11e8-80d7-ec35c52c28e2.png)

Is this expected? I thought the logo was on the right based on what we looked at last week. 

Thanks for doing all of this!!!

julientregoat commented:
Yeah, I was playing with the positioning of the logo, going to switch it back because it doesn't look right. Also I think I was a bit tired when I was playing with the margins because it should nooooooooot be like that! Here's a screenshot of what I'm pushing up now

![screeenshot-localhost](https://user-images.githubusercontent.com/33069092/38789819-cecc7a5a-410a-11e8-9c29-3f155a2ddded.png)

prestonvanloon commented:
Sounds good @julientregoat. Raul and I will review when you've finished the first goal on your plan.

Thanks!

prestonvanloon commented on docs/CNAME:
```diff
@@ -0,0 +1 @@
+prysmaticlabs.com
```
I think you need to delete the old CNAME since this directory will be the new root

prestonvanloon commented on src/App.css:
```diff
@@ -0,0 +1,322 @@
+  font-family: 'Oxygen';
+}
+
+/* main page */
```
Can we keep the styles within the component? This file is 322 lines for global styles, but many of these classes only apply to a specific component.

prestonvanloon commented on src/FAQ/FAQPage.js:
```diff
@@ -0,0 +1,17 @@
+    <div className="faq-page">
+      <h2 className="yel">FAQ</h2>
+      <p><span className="pink">Q:</span> Is this a for-profit venture? </p>
+  		<p><span className="blue">A:</span> No, not at the moment. We are a team of open-source contributors implementing <a className="pink" href="https://github.com/ethereum/wiki/wiki/Sharding-FAQ">sharding</a> for Ethereum because we truly believe the protocol desperately needs this ASAP. As a team, we may consider a for profit venture in the future, but we do not intend to profit directly from ou ...
```
Can you split this into multiple lines so it conforms to a reasonable column limit? Maybe 100 characters?

prestonvanloon commented on src/FAQ/FAQPage.js:
```diff
@@ -0,0 +1,17 @@
+  		<p><span className="pink">Q:</span> How can I contribute? </p>
+  		<p><span className="blue">A:</span> Talk to us on <a className="pink" href="https://gitter.im/prysmaticlabs/geth-sharding?source=orgpage">Gitter</a>! We are always looking for more contributors. We'll guide you on how to submit pull requests from there. You can also check our current progress on our Github page.</p>
+  		<p><span className="pink">Q:</span> How can I stay updated? </p>
+  		<p><span className="blue">A:</span> Join our <a className="pink" href="https://rauljordan.us14.list-manage.com/subscribe?u=7bcfcec175d2461e541a3f2a8&id=3a2a17337c">mailing list</a>! We will also be posting major milestones on the <a className="pink" href="https://www.reddit.com/r/ethereum/">/r/ethereum</a> Reddit. We also provide updates on <a className="pink" href="https://www.youtube.com/ch ...
```
Here too please

prestonvanloon commented on src/Footer.js:
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Footer = (props) => {
```
Can we put this in it's own component directory or group directory for common components?

https://reactjs.org/docs/faq-structure.html#grouping-by-features-or-routes

prestonvanloon commented on src/logos/Prysm.svg:
```diff
@@ -0,0 +1,49 @@
+<svg width="153" height="173" viewBox="0 0 153 173" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Can the logos go in assets? maybe `src/assets/logos/...`

prestonvanloon commented on src/App.css:
```diff
@@ -0,0 +1,322 @@
+  border-bottom: 4px solid #eb4a9d;
+  border-bottom-right-radius: 5px;
+  border-bottom-left-radius: 5px;
+  animation: slide-in 1.3s ease-in-out both;
```
Can we leave out animations? I think it's a bit distracting or overwhelming on page load.

prestonvanloon commented on src/App.css:
```diff
@@ -0,0 +1,322 @@
+.team-container {
+  margin-left: 1%;
+  margin-right: 1%;
+  animation: slide-in 1.3s ease-in-out 0.75s both;
```
Here too, I think these big animations take away from the page or are distracting.

rauljordan commented on src/Main/TeamContainer.js:
```diff
@@ -0,0 +1,18 @@
+
+const TeamContainer = (props) => {
+  return (
+    <div className="team-container">
```
The font-size of this component seems a bit too small compared to the text above it. Can we make a bit bigger?

rauljordan commented on src/Main/Intro.js:
```diff
@@ -0,0 +1,24 @@
+    <div className="intro">
+      <h2>A team of blockchain engineers committed to scaling Ethereum.</h2>
+      <p>We are implementing the first <a className="pink" href="https://github.com/ethereum/wiki/wiki/Sharding-FAQ">sharding</a> client for the go-ethereum project.</p>
+      <p>Currently, blockchains do not scale. With Ethereum receiving over <i>a million transactions per day</i>, the network is desperate for a solution that will help sustain its massive growth and adoption.</p> 
```
Instead of italics, can we bold a million transactions per day or instead just change the color of it? the italics look a bit weird

rauljordan commented on src/Main/Intro.js:
```diff
@@ -0,0 +1,24 @@
+
+const Intro = (props) => {
+  return (
+    <div className="intro">
```
Can we make the line height of the text in this component bigger? Seems too close together

rauljordan commented on src/Main/IntroContainer.js:
```diff
@@ -0,0 +1,15 @@
+  return (
+  	<div className="intro-container">
+  		<Intro />
+    	<BigLogo />
```
Distance between big logo and intro container should be bigger. Perhaps we can also adjust the margins of the entire page container to be smaller than they are now?

terenc3t commented:
This looks great! thanks a lot for the help! @julientregoat

julientregoat commented:
Okay, I've implemented the changes as requested. 

@prestonvanloon, I modularized the CSS files to a per container-component basis and cleaned up the file structure, as well as cleaned up column widths. I also removed the page load slide in animations. If it ends up looking a bit too static for your taste, I can put in something a bit more subtle. 

@rauljordan  I wasn't sure if you had a preference as to how far in you'd like the content under the borders, so I adjusted both the overall app margins as well as the container margins evenly to give a bit more space between the intro and big logo. I also increased the line height a bit as well, definitely looks way more readable. I kind of want to get the intro header line on one line, but it would either be smaller or be a bit closer to the logo. Let me know what you think

prestonvanloon commented on src/about/AboutPage.js:
```diff
@@ -0,0 +1,15 @@
+  return (
+    <div> 
+    	<IntroContainer />
+		<TeamContainer />
```
Formatting

prestonvanloon commented on src/about/Intro.js:
```diff
@@ -0,0 +1,41 @@
+            Join Our Gitter Channel <img src={Gitter} alt="gitter icon"/>
+          </button>
+        </a>
+  		</div>
```
Formatting

prestonvanloon commented on src/about/TeamMember.js:
```diff
@@ -0,0 +1,17 @@
+const TeamMember = ({ member }) => {
+  return (
+  	<div className="team-member">	
+  		<a href={member.github}>
```
Formatting

prestonvanloon commented:
A general comment: DO NOT submit until we reconfigure the site to serve from `docs/`.

julientregoat commented:
Okay, the formatting issues appear to be solved. Getting used to a new code editor so I had spaces and tabs intermingling, which didn't translate well outside of it. Everything should be formatted properly now. I will work with the esformatter-jsx plugin as well to see what that turns out.
